lib/resty/websocket/protocol.lua

```diff
@@ -175,13 +177,12 @@ function _M.recv_frame(sock, max_payload_len, force_masking)
 
                 if payload_len > 2 then
                     -- TODO string.buffer optimizations
-                    local bytes = new_tab(payload_len - 2, 0)
+                    local bytes = ffi_new("char[?]", payload_len - 2)
```

Comment:
we can introduce a `get_string_buff` function here, just like it in `resty.core.base`.
https://github.com/openresty/lua-resty-core/blob/master/lib/resty/core/base.lua#L182

so that we can avoid creating a new cdata object when the `payload_len` is not too big.(like 4096 or 8192).

Reply:
@doujiang24  After following your guide, still taking 1 million times benchmark, the `get_string_buff` implementation improves 10%+ performance than always calling `ffi_new` implementation. Here is the [benchmark code ](https://gist.github.com/Gerrard-YNWA/ac0af5939c90072beae33995f720549a), and the result is 
```
resty test.lua
=========benchmark1 cost:52486.999988556 ms.
=========benchmark2 cost:5371.0000514984 ms.
=========benchmark3 cost:4767.9998874664 ms.
```
Thanks a lot!